baseplate/lib/circuit_breaker/factory.py

```diff
@@ -0,0 +1,164 @@
+import logging
+
+from contextlib import contextmanager
+from datetime import timedelta
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import Tuple
+from typing import Type
+
+from baseplate import Span
+from baseplate.clients import ContextFactory
+from baseplate.lib import config
+from baseplate.lib.circuit_breaker.breaker import Breaker
+from baseplate.lib.circuit_breaker.breaker import BreakerState
+from baseplate.lib.circuit_breaker.errors import BreakerTrippedError
+
+
+logger = logging.getLogger(__name__)
+
+
+class CircuitBreakerClientWrapperFactory(ContextFactory):
+    """Provide an object combining a client and circuit breaker for use with the client.
+
+    When attached to the baseplate `RequestContext` can be used like:
+
+    ```
+    breakable_exceptions = (...)  # exceptions indicating the service is unhealthy
+    with context.breaker_wrapped_client.breaker_context("identifier", breakable_exceptions) as svc:
+        svc.get_something()
+    ```
+    """
+
+    def __init__(self, client_factory: ContextFactory, breaker_box: "CircuitBreakerBox"):
+        self.client_factory = client_factory
+        self.breaker_box = breaker_box
```

Comment:
so in this way, people will have to initialize the breaker_box in. __init__ file and pass it in to factory?

Reply:
yeah exactly. that's similar to how we do other clients in baseplate.py, and will make sure that all workers/greenlets share the same breakers.